be/src/exprs/vectorized/math_functions.cpp

```diff
@@ -104,6 +115,36 @@ DEFINE_UNARY_FN_WITH_IMPL(ZeroCheck, value) {
                 VECTORIZED_FN_ARGS(0), VECTORIZED_FN_ARGS(1));                                                \
     }
 
+#define DEFINE_MATH_TRUNCATE_BINARY_FN_WITH_IMPL(NAME, LTYPE, RTYPE)                                         \
+    template <PrimitiveType RESULT_PT>                                                                       \
+    struct NAME##Impl {                                                                                      \
+        template <typename LType, typename RType, typename ResultType>                                       \
+        static inline ResultType apply(const LType& l, const RType& r, const ColumnPtr& l_column,            \
+                                       const ColumnPtr& r_column) {                                          \
+            return Truncate<LTYPE, RESULT_PT>::truncate(l, r, l_column, r_column);                           \
+        }                                                                                                    \
+    };                                                                                                       \
+    ColumnPtr MathFunctions::NAME(FunctionContext* context, const starrocks::vectorized::Columns& columns) { \
+        static_assert(pt_is_decimal<LTYPE>, "LTYPE must be Decimal");                                        \
+        const auto& type = context->get_return_type();                                                       \
+        if (type.precision <= 9 && LTYPE == PrimitiveType::TYPE_DECIMAL32) {                                 \
```

Comment:
What's `9`?

Reply:
It means the maximum precision of decimal
* decimal32 -> 9
* decimal64->18
* decimal128->38

I'll add some comments